provider/alibabacloud/alibaba_cloud.go

```diff
@@ -661,20 +661,41 @@ func (p *AlibabaCloudProvider) updateRecords(recordMap map[string][]alidns.Recor
 
 func (p *AlibabaCloudProvider) splitDNSName(fullName string) (rr string, domain string) {
 	name := strings.TrimSuffix(fullName, ".")
-	parts := strings.Split(name, ".")
-	if len(parts) < 2 {
-		rr = name
-		domain = ""
-	} else {
-		domain = parts[len(parts)-2] + "." + parts[len(parts)-1]
-		rrIndex := strings.Index(name, domain)
-		if rrIndex < 1 {
-			rrIndex = 1
+
+	var found bool
+	domainNames, tmpErr := p.getDomainList()
+	if tmpErr != nil {
+		log.Errorf("AlibabaCloudProvider getDomainList error %v", tmpErr)
+	}
+	for _, tmpDomainName := range domainNames {
+		if strings.HasSuffix(name, "."+tmpDomainName) {
+			rr = name[0 : len(name)-len(tmpDomainName)-1]
+			domain = tmpDomainName
+			found = true
+			break
```

Comment:
Why does this have a `break` but  the `name == tmpDomainName` not?

It seems the order of domains is significant. Should it be?